eppoclient/client.go

```diff
@@ -60,86 +77,35 @@ func (ec *EppoClient) getAssignment(subjectKey string, flagKey string, subjectAt
 		panic("no flag key provided")
 	}
 
-	config, err := ec.configRequestor.GetConfiguration(flagKey)
+	flag, err := ec.configRequestor.GetConfiguration(flagKey)
 	if err != nil {
-		return Null(), err
+		return nil, err
 	}
 
-	override := getSubjectVariationOverride(config, subjectKey, valueType)
-	if override != Null() {
-		return override, nil
-	}
-
-	// Check if disabled
-	if !config.Enabled {
-		return Null(), errors.New("the experiment or flag is not enabled")
-	}
-
-	// Find matching rule
-	rule, err := findMatchingRule(subjectAttributes, config.Rules)
+	err = flag.verifyType(variationType)
```

Comment:
In swift (and android and python) we had to also handle when the variation type did not match the desired kind. specifically the `invalid-value-flag` test where an `INTEGER` variation type is desired but a double is received. does that get handled for you in a way i'm not seeing? maybe `valueToAssignmentValue` is handling that

Reply:
Yes. That's handled in `valueToAssignmentValue`: https://github.com/Eppo-exp/golang-sdk/pull/41/files/80ac9ea3cf91ce0e12a64b97526135d674f4b16d#diff-05660c7f68f1998db7bb1977133831b136836ad85000bedcf3bd0917aee448f6R89

We can't handle this verification here yet as it depends on which variant is selected